R/create.autoencoder.R

```diff
@@ -8,12 +8,12 @@ create.autoencoder <- function(
 	) {
 
 	# input checks
-	if (class(data.matrix) != 'matrix') {
-		stop('data.matrix needs to be a matrix');
+	if (!("matrix" %in% class(data.matrix) || "data.frame" %in% class(data.matrix))) {
+		stop('data.matrix needs to be a matrix or a data frame')
```

Comment:
@tyamaguchi-ucla 
Sorry to bother you again. So I update this from the original check script, because @jarbet found out the example is not working #11 
Any concern with this? If not, I will change the check script in other functions.

Reply:
I'm not sure why you had to use the `%in%` operator. If you want to use the `%in%` operator,
`!(class(data.matrix) %in% c('matrix', 'data.frame'))` make more sense to me or we can use the logic that you used in the other scripts.

Reply:
> I'm not sure why you had to use the `%in%` operator. If you want to use the `%in%` operator, `!(class(data.matrix) %in% c('matrix', 'data.frame'))` makes more sense to me or we can use the logic that you used in the other scripts.

The issue is `class(data.matrix)` can return a vector of multiple classes.  So your suggestion would need to be modified a little:

```
!(any(class(data.matrix) %in% c('matrix, 'data.frame')));
```

Reply:
@jarbet Good point!

Reply:
@jarbet It seems that we probably want to go with `is.matrix()` and `is.data.frame()` instead.